### PR TITLE
Initial project structure

### PR DESCRIPTION
**Pull request by JSiuda1**

### CODE REVIEW

File: .clang-format
Comment (kgliwinski) [inline .clang-format:1]: To add:
InsertNewlineAtEOF: true **(this is must have - limts lint issues, but doesn't eliminate them)**
BreakBeforeBraces: Attach
InsertBraces: true

File: .github/workflows/gitci.yml
Comment (kgliwinski) [inline .github/workflows/gitci.yml:1]: Nice to have if tests are planned:
```yml
on: push

jobs:
  build_ESP32:

    runs-on: ubuntu-latest

    steps:
    - name: Checkout repo
      uses: actions/checkout@v2
      with:
        submodules: 'recursive'
    - name: esp-idf build
      uses: espressif/esp-idf-ci-action@v1
      with: 
        esp_idf_version: release-v5.1
        target: 'esp32s3'
        path: 'Mainboard_ESP32'
        
  build_ESP32_tests:
      
      runs-on: ubuntu-latest
  
      steps:
      - name: Checkout repo
        uses: actions/checkout@v2
        with:
          submodules: 'recursive'
      - name: esp-idf build
        uses: espressif/esp-idf-ci-action@v1
        with: 
          esp_idf_version: release-v5.1
          target: 'esp32s3'
          path: 'Mainboard_ESP32/test'

  lint:
    runs-on: ubuntu-latest


    steps:
    - name: Checkout repo
      uses: actions/checkout@v2

    - name: Set up Python
      uses: actions/setup-python@v4

    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        pip install cpplint
    - name: run lint
      run: bash lint.sh
```

File: .clang-format
Comment (kgliwinski) [reply .clang-format:1]: The rest of these are optional

**Review by kgliwinski — APPROVED** [review]
Review comments before merging 🙏 🐕

File: .clang-format
Comment (JSiuda1) [reply .clang-format:1]: G

File: .github/workflows/gitci.yml
Comment (JSiuda1) [reply .github/workflows/gitci.yml:1]: Cpplint and cppcheck have been configured :nerd_face:

File: CPPLINT.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+filter=-build/include_subdir
+linelength=80
```
Comment (kgliwinski) [inline CPPLINT.cfg]: ColumnLimit in clang-format set to 100, either change this to 100 or ColumnLimit to 80

**Review by kgliwinski — CHANGES_REQUESTED** [review]

**Review by kgliwinski — APPROVED** [review]
Essa

**Pull request merged**